test/dataset/test_sampling.py

```diff
@@ -40,6 +40,11 @@ def test_single_cut_sampler_shuffling():
     # Invariant 3: the items are shuffled, i.e. the order is different than that in the CutSet
     assert sampler_cut_ids != [c.id for c in cut_set]
 
+    # check that len(sampler) gives consistent result
```

Comment:
I think this test's inputs are insufficient to test this, as they all have equal lengths

Reply:
But that should be the point no? The lengths should equal

Reply:
sorry I didn't express myself well... I meant that the cut inputs have equal lengths, so whatever is their order inside the sampler, you will always get the same len(sampler) with your changes, regardless of the epoch. If we create cut inputs with varying duration, then you can actually test whether len(sampler) is consistent in the same epoch and different in different epochs

Reply:
Yup I figured in the end :) The updated version uses varying lengths.